prow/plugins/size/gen_files_group.go

```diff
@@ -0,0 +1,188 @@
+		FilePrefixes: make(map[string]bool),
+	}
+
+	var bs []byte
+	bs, err = gc.GetFile(owner, repo, genConfigFile, sha)
+	if err != nil {
+		err = &IOError{err: fmt.Errorf("GetFile error: %v", err)}
+		return
+	}
+
+	var repoFiles []string
+	repoFiles, parseErrs, err = g.load(bytes.NewBuffer(bs))
+	if err != nil {
+		return
+	}
+	for _, f := range repoFiles {
+		bs, err = gc.GetFile(owner, repo, f, sha)
+		if err != nil {
+			return
+		}
+		if err = g.loadPaths(bytes.NewBuffer(bs)); err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Use load to read a generated files config file, and populate g with the commands.
+// "paths-from-repo" commands are aggregated into repoPaths. It is the caller's
+// responsiblity to fetch these and load them via g.loadPaths.
+func (g *GenFilesGroup) load(r io.Reader) (repoPaths []string, parseErrs []*ParseError, err *IOError) {
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		l := strings.TrimSpace(s.Text())
+		if l == "" || l[0] == '#' {
+			// Ignore comments and empty lines.
+			continue
+		}
+
+		fs := strings.Fields(l)
+		if len(fs) != 2 {
+			parseErrs = append(parseErrs, &ParseError{line: l})
+			continue
+		}
+
+		switch fs[0] {
+		case "prefix", "path-prefix":
+			g.PathPrefixes[fs[1]] = true
+		case "file-prefix":
+			g.FilePrefixes[fs[1]] = true
+		case "file-name":
+			g.FileNames[fs[1]] = true
+		case "path":
+			g.FileNames[fs[1]] = true
+		case "paths-from-repo":
+			// Despite the name, this command actually requires a file
+			// of paths from the _same_ repo in which the .generated_files
+			// config lives.
+			repoPaths = append(repoPaths, fs[1])
+		default:
+			parseErrs = append(parseErrs, &ParseError{line: l})
+		}
+	}
+	if err := s.Err(); err != nil {
+		err = &IOError{err: err}
+	}
+	return
+}
+
+// Use loadPaths to load a file of new-line delimited paths, such as
+// resolving file data referenced in a "paths-from-repo" command.
+func (g *GenFilesGroup) loadPaths(r io.Reader) *IOError {
+	s := bufio.NewScanner(r)
+
+	for s.Scan() {
+		l := strings.TrimSpace(s.Text())
+		if l == "" || l[0] == '#' {
+			// Ignore comments and empty lines.
+			continue
+		}
+
+		g.Paths[l] = true
+	}
+
+	if err := s.Err(); err != nil {
+		return &IOError{
+			err: err,
+		}
+	}
+
+	return nil
+}
+
+// Use Match to determine whether a file, given here by its full path
+// is included in the generated files group.
+func (g *GenFilesGroup) Match(path string) bool {
+	// TODO(nlandolfi): could make this faster with a better structure
+	// might not be worth it...tries...
+
+	if g.Paths[path] {
+		return true
+	}
+
+	for prefix := range g.PathPrefixes {
+		if strings.HasPrefix(path, prefix) {
+			return true
+		}
+	}
+
+	base := filepath.Base(path)
+
+	if g.FileNames[base] {
+		return true
+	}
+
+	for prefix := range g.FilePrefixes {
+		if strings.HasPrefix(base, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IOError is related to reading or writing bytes.
+type IOError struct {
```

Comment:
This is probably unnecessary. We can just assume that if it's not a `ParseError` then it's something to worry about.